tests/Unit/Extensions/ColorsExtensionTest.php

```diff
@@ -0,0 +1,392 @@
+<?php
+
+namespace Xefi\Faker\Tests\Unit\Extensions;
+
+use Random\Randomizer;
+use Xefi\Faker\Container\Container;
+use Xefi\Faker\Extensions\ColorsExtension;
+
+final class ColorsExtensionTest extends TestCase
+{
+    protected array $safeColorNames = [];
+    protected array $colorNames = [];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $colorExtension = new ColorsExtension(new Randomizer());
+        $this->safeColorNames = (new \ReflectionClass($colorExtension))->getProperty('safeColorNames')->getValue($colorExtension);
+        $this->colorNames = (new \ReflectionClass($colorExtension))->getProperty('colorNames')->getValue($colorExtension);
+    }
+
+    public function testSafeColorName(): void
+    {
+        $faker = new Container(false);
+
+        $results = [];
+        for ($i = 0; $i < count($this->safeColorNames); $i++) {
+            $results[] = $faker->unique()->safeColorName();
+        }
+
+        $this->assertEqualsCanonicalizing(
+            $this->safeColorNames,
+            $results
+        );
+    }
+
+    public function testColorName(): void
+    {
+        $faker = new Container(false);
+
+        $results = [];
+        for ($i = 0; $i < count($this->colorNames); $i++) {
+            $results[] = $faker->unique()->colorName();
+        }
+
+        $this->assertEqualsCanonicalizing(
+            $this->colorNames,
+            $results
+        );
+    }
+
+    public function testSafeHexColor(): void
+    {
+        $faker = new Container(false);
+
+        for ($i = 0; $i < 100; $i++) {
+            $color = $faker->unique()->safeHexColor();
+
+            $this->assertStringStartsWith('#', $color);
+            $this->assertTrue(ctype_xdigit(substr($color, 1)));
+            $this->assertEquals(7, strlen($color));
+        }
+    }
+
+    public function testHexColor(): void
+    {
+        $faker = new Container(false);
```

Comment:
We might need to put this on the base "testCase" class, we use it every where, should be easier to use "$this->faker"